catalog/rocks/catalog.go

```diff
@@ -104,17 +113,50 @@ type EntryRecord struct {
 
 // Commit represents commit metadata (author, time, tree ID)
 type Commit struct {
-	Committer    string
-	Message      string
-	TreeID       TreeID
-	CreationDate time.Time
-	Parents      []CommitID
-	Metadata     map[string]string
+	Committer    string           `db:"committer"`
+	Message      string           `db:"message"`
+	TreeID       TreeID           `db:"tree_id"`
+	CreationDate time.Time        `db:"creation_date"`
+	Parents      CommitParents    `db:"parents"`
+	Metadata     catalog.Metadata `db:"metadata"`
+}
+
+func (c Commit) ID() CommitID {
+	// TODO(ozkatz): move the hashing logic and encoding to a module others can use
+	h := sha256.New()
+	h.Write([]byte(c.Committer))
+	h.Write([]byte{0})
```

Comment:
I prefer to use length-prefixed writes.  Otherwise we get another illegal rune in most fields.  Or... AFAIK `git` just hashes its file contents, you could decide on a single textual format, stream a commit into that format, and then pass it `h`.

Reply:
yea, this requires a redo. we need a canonical way of doing stuff like that that works regardless of entity type (same as we had before MVCC with an `Identifiable` interface that defines an `Identity() []byte` method